src/chart/Column.js

```diff
@@ -23,61 +23,91 @@
 
         this._palette = this._palette.switch(this.paletteID());
 
-        var column = this.svgData.selectAll(".columnRect")
+        var dataLen = 10;
+        switch (this.xAxisType()) {
+            case "ordinal":
+              dataLen = this.dataScale.rangeBand();
+               break;
+            case "linear":
+            case "time":
+               dataLen = Math.max(Math.abs(this.dataScale(2) - this.dataScale(1)) * (100 - this._linearGap) / 100, 1);
+               break;
+        }
+
+        var columnScale = d3.scale.ordinal()
+            .domain(context._columns.filter(function (d, idx) { return idx > 0; }))
+            .rangeRoundBands([0, dataLen])
+        ;
+
+        var column = this.svgData.selectAll(".dataRow")
             .data(this.data())
         ;
 
-        var title = column
-          .enter().append("rect")
-            .attr("class", "columnRect")
-            .on("click", function (d) {
-                context.click(context.rowToObj(d), context._columns[1]);
-            })
-            .append("title")
+        column.enter().append("g")
+            .attr("class", "dataRow")
         ;
 
-        column.transition()
-            .style("fill", function (d) { return context._palette(d[0]); })
-            .each(function (d) {
+        column
+            .each(function (dataRow, i) {
+                var dataPos = context.dataScale(dataRow[0]);
                 var element = d3.select(this);
-                var dataPos = context.dataScale(d[0]);
-                var dataLen = 10;
+                var offset = 0;
```

Comment:
The offset can be calculated inside the same switch statement that the dataLen is calculated.